test/plain.jl

```diff
@@ -412,3 +412,20 @@ close(f)
 rm(fn)
 
 end # testset null and undefined
+
+# test writing reinterpreted data
+@testset "reinterpreted data" begin
+fn = tempname()
+
+h5open(fn, "w") do f
+    data = reinterpret(UInt8, [true, false, false])
+    write(f, "reinterpret array", data)
+end
+
+@test h5open(fn, "r") do f
+    read(f, "reinterpret array")
+end == UInt8[0x01, 0x00, 0x00]
+
+rm(fn)
```

Comment:
Not necessary but it's nice to do:
```julia
fn = tempname()
try
    ...
finally
    rm(fn)
end
```
to ensure the file gets deleted even when an exception occurs